src/main/java/storage/Storage.java

```diff
@@ -306,4 +308,67 @@ public boolean renameModule(String newModuleName, Module module) {
         boolean success = file.renameTo(new File(getFilePath() + "/" + newModuleName));
         return success;
     }
+
+    public void createHistoryDir() {
+        File f = new File(filePath + "/" + "history");
+        boolean historyDirExists = f.exists();
+        if (!historyDirExists) {
+            f.mkdir();
+        }
+    }
+
+    public void createHistory(String date) {
+        try {
+            File f = new File(filePath + "/history/" + date + ".txt");
+            boolean historyFileExists = f.exists();
+            boolean historyFileCreated = false;
+            if (!historyFileExists) {
+                historyFileCreated = f.createNewFile();
+            }
+
+            if (historyFileCreated) {
+                System.out.println("    Successfully created new history file " + date + ".txt");
+            }
+        } catch (IOException e) {
+            System.out.println("Error creating the file.");
```

Comment:
Perhaps you could pass ui from Kaji to this method and call ui.showToUser and ui.showError instead of using System.out.println?